pkg/api/backuptarget/healthy_handler.go

```diff
@@ -0,0 +1,80 @@
+package backuptarget
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	"github.com/longhorn/backupstore"
+	"github.com/pkg/errors"
+	ctlcorev1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+
+	"github.com/harvester/harvester/pkg/config"
+	"github.com/harvester/harvester/pkg/controller/master/backup"
+	"github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/settings"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+type HealthyHandler struct {
+	context      context.Context
+	settingCache v1beta1.SettingCache
+	secretCache  ctlcorev1.SecretCache
+}
+
+func NewHealthyHandler(scaled *config.Scaled) *HealthyHandler {
+	return &HealthyHandler{
+		context:      scaled.Ctx,
+		settingCache: scaled.HarvesterFactory.Harvesterhci().V1beta1().Setting().Cache(),
+		secretCache:  scaled.CoreFactory.Core().V1().Secret().Cache(),
+	}
+}
+
+func (h *HealthyHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	backupTargetName := mux.Vars(r)["backupTargetName"]
+	if backupTargetName != "default" {
+		util.ResponseError(rw, http.StatusBadRequest, fmt.Errorf("currently, only support default backup target"))
+		return
+	}
+
+	backupTargetSetting, err := h.settingCache.Get(settings.BackupTargetSettingName)
+	if err != nil {
+		util.ResponseError(rw, http.StatusInternalServerError, errors.Wrapf(err, "can't get %s setting", settings.BackupTargetSettingName))
+		return
+	}
+	if backupTargetSetting.Value == "" {
+		util.ResponseError(rw, http.StatusBadRequest, fmt.Errorf("%s setting is not set", settings.BackupTargetSettingName))
+		return
+	}
+
+	target, err := settings.DecodeBackupTarget(backupTargetSetting.Value)
+	if err != nil {
+		util.ResponseError(rw, http.StatusInternalServerError, errors.Wrapf(err, "can't decode %s setting %s", settings.BackupTargetSettingName, backupTargetSetting.Value))
+		return
+	}
+	if target.IsDefaultBackupTarget() {
+		util.ResponseError(rw, http.StatusBadRequest, fmt.Errorf("%s setting is not set", settings.BackupTargetSettingName))
+		return
+	}
+
+	if target.Type == settings.S3BackupType {
+		secret, err := h.secretCache.Get(util.LonghornSystemNamespaceName, util.BackupTargetSecretName)
+		if err != nil {
+			util.ResponseError(rw, http.StatusInternalServerError, errors.Wrapf(err, "can't get %s/%s secret", util.LonghornSystemNamespaceName, util.BackupTargetSecretName))
+			return
+		}
+		os.Setenv(backup.AWSAccessKey, string(secret.Data[backup.AWSAccessKey]))
+		os.Setenv(backup.AWSSecretKey, string(secret.Data[backup.AWSSecretKey]))
+		os.Setenv(backup.AWSEndpoints, string(secret.Data[backup.AWSEndpoints]))
+		os.Setenv(backup.AWSCERT, string(secret.Data[backup.AWSCERT]))
+	}
+
+	_, err = backupstore.GetBackupStoreDriver(backup.ConstructEndpoint(target))
+	if err != nil {
+		util.ResponseError(rw, http.StatusServiceUnavailable, errors.Wrapf(err, "can't connect to backup target %+v", target))
```

Comment:
always return the error message, don't swallow it.

Reply:
After reading this thread, I change to use `fmt.Errorf("... %w", err)`. Thank you.

https://stackoverflow.com/questions/61933650/whats-the-difference-between-errors-wrapf-errors-errorf-and-fmt-errorf